### PR TITLE
[ssr] check cleared hyps do exist (fix #7050)

### LINKED ISSUE #7050: ssr: no error message when removing non-existing assumption with a clear switch
#### Version

Coq 8.8+alpha (ef0fc77)

#### Operating system

macOS 10.13.3 High Sierra

#### Description of the problem

Coq does _not_ complain about the undefined name in the clear switch: 

```coq
Example foo : True -> True.
Proof.
move=> {NO_SUCH_NAME} _.
Abort.
```

Coq 8.7.2 outputs an error message in this case:
```
> move=> {NO_SUCH_NAME} _.
> ^^^^^^^^^^^^^^^^^^^^^^^
Error:
Ltac call to "move (ssrmovearg) (ssrclauses)" failed.
No assumption is named NO_SUCH_NAME
```

### PR DESCRIPTION
**Pull request by gares**
It is a regression only wrt 8.7 (so if we fix it in time for 8.8) it won't be one.

Let's see how it goes with CI, I tried to preserve the v8.7 semantics.

### CODE REVIEW

File: plugins/ssr/ssripats.ml [hunk 1]
```diff
@@ -133,6 +133,12 @@ let intro_clear ids future_ipats =
     isCLR_PUSHL clear_ids
 end
 
+let tacCHECK_HYPS_EXIST hyps = Goal.enter begin fun gl ->
+  let ctx = Goal.hyps gl in
+  List.iter (Ssrcommon.check_hyp_exists ctx) hyps;
```
Comment (maximedenes) [inline plugins/ssr/ssripats.ml]: Are you sure raising an error is what we want? I forgot, what is the recommended way to have a tactic fail @ppedrot ?

Comment (maximedenes) [reply plugins/ssr/ssripats.ml]: Ok, I don't see why this exception catching is correlated to focusing on a goal, but I should get a tactic monad crash course before asking questions.

**Review by maximedenes — APPROVED** [review]

File: plugins/ssr/ssripats.ml (hunk 1 above)
Comment (ppedrot) [reply plugins/ssr/ssripats.ml]: For historical reasons, `Goal.enter` was made to catch exceptions. This was done in order to ease the porting of non-monadic code, as a typical pattern would have been going from
```
fun gl -> blah
```
to
```
enter begin fun gl -> blah end
```
where `blah` would typically raise exceptions to signal failure. This pattern is still used a lot in the current code, even though it's a bit fragile, as one needs to be careful not to add closures in the surrounding context of exception raise.

**Comment by gares** [comment]
@maximedenes this one is ready

**Pull request merged**